File: sdk/SDK.Examples/src/GetPackageExample.cs
using System;
using System.IO;
using Silanis.ESL.SDK;
using Silanis.ESL.SDK.Builder;

namespace SDK.Examples
{
	public class GetPackageExample : SDKSample
	{
        public static void Main (string[] args)
        {
            new GetPackageExample().Run();
        }

        override public void Execute()
        {
            DocumentPackage package = PackageBuilder.NewPackageNamed (PackageName)
					.DescribedAs ("This is a new package")
					.WithSigner(SignerBuilder.NewSignerWithEmail(email1)
					            .WithFirstName("John")
					            .WithLastName("Smith")
					            .WithCompany ("Acme Inc")
					            .WithTitle ("Managing Director"))
					.WithDocument(DocumentBuilder.NewDocumentNamed("My Document")
                                  .FromStream(fileStream1, DocumentType.PDF)
					              .WithSignature(SignatureBuilder.SignatureFor(email1)
					              		.OnPage(0)
					               		.AtPosition(500, 100)
					               		.WithField(FieldBuilder.SignatureDate()
					           				.OnPage(0)
					           				.AtPosition(500, 200))
					               		.WithField(FieldBuilder.SignerName ()
					           				.OnPage(0)
					           				.AtPosition(500, 300))
							         	.WithField(FieldBuilder.SignerTitle()
							           		.OnPage(0)
							           		.AtPosition(500, 400))
					               		.WithField (FieldBuilder.SignerCompany()
					            			.OnPage (0)
					            			.AtPosition (500, 500))))
					.Build ();

			PackageId id = eslClient.CreatePackage (package);
			eslClient.SendPackage(id);

			DocumentPackage retrievedPackage = eslClient.GetPackage (id);
            Console.WriteLine("Document retrieved = " + retrievedPackage.Id);
		}
	}
}

File: sdk/SDK.VBExamples/SDK.VBExamples/GetPackageExample.vb
Imports System
Imports System.IO
Imports Silanis.ESL.SDK
Imports Silanis.ESL.SDK.Builder

Namespace SDK.VBExamples
    Public Module GetPackageExample
        Dim apiToken = "UGF3anRiZHJxVUNNOnNlY3JldA=="
        Dim baseUrl = "http://localhost:8080"

        Sub Main()
            ' Create new esl client with api token and base url
            Dim client = New EslClient(apiToken, baseUrl)
            Dim file = New FileInfo(Directory.GetCurrentDirectory() + My.Resources.document)

            Dim package = PackageBuilder.NewPackageNamed("Fields example " + DateTime.Now) _
                .DescribedAs("This is a new package") _
                .WithSigner(SignerBuilder.NewSignerWithEmail("john.smith@email.com") _
                    .WithFirstName("John") _
                    .WithLastName("Smith") _
                    .WithCompany("Acme Inc") _
                    .WithTitle("Managing Director")) _
                .WithDocument(DocumentBuilder.NewDocumentNamed("My Document") _
                    .FromFile(file.FullName) _
                    .WithSignature(SignatureBuilder.SignatureFor("john.smith@email.com") _
                        .OnPage(0) _
                        .AtPosition(500, 100) _
                        .WithField(FieldBuilder.SignatureDate() _
                            .OnPage(0) _
                            .AtPosition(500, 200)) _
                        .WithField(FieldBuilder.SignerName() _
                            .OnPage(0) _
                            .AtPosition(500, 300)) _
                        .WithField(FieldBuilder.SignerTitle() _
                            .OnPage(0) _
                            .AtPosition(500, 400)) _
                        .WithField(FieldBuilder.SignerCompany() _
                            .OnPage(0) _
                            .AtPosition(500, 500)))) _
                .Build()

            Dim id = client.CreatePackage(package)
            client.SendPackage(id)

            Console.WriteLine("Package {0} was sent", id.Id)

            Dim retrievedPackage = client.GetPackage(id)

            Console.WriteLine("id = {0}", retrievedPackage.Id)
            Console.WriteLine("status = {0}", retrievedPackage.Status)
        End Sub
    End Module
End Namespace